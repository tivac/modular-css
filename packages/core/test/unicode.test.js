"use strict";

var namer  = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("unicode", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer
            });
        });

        it("should support unicode classes & ids", () =>
            processor.file(
                require.resolve("./specimens/unicode.css")
            )
            .then(() => processor.output({
                to : "./packages/core/test/output/unicode.css"
            }))
            .then((output) =>
                expect(output.css).toMatchSnapshot()
            )
        );
    });
});
