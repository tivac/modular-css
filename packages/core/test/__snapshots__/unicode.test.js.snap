// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js unicode should support unicode classes & ids 1`] = `
"/* packages/core/test/specimens/unicode.css */
#♥ { background: lime; }
#© { background: lime; }
#“‘’” { background: lime; }
#☺☃ { background: lime; }
#⌘⌥ { background: lime; }
#𝄞♪♩♫♬ { background: lime; }
#💩 { background: lime; }
#\\\\? { background: lime; }
#\\\\@ { background: lime; }
#\\\\. { background: lime; }
#\\\\3A \\\\) { background: lime; }
#\\\\3A \\\\\`\\\\( { background: lime; }
#\\\\31 23 { background: lime; }
#\\\\31 a2b3c { background: lime; }
#\\\\<p\\\\> { background: lime; }
#\\\\<\\\\>\\\\<\\\\<\\\\<\\\\>\\\\>\\\\<\\\\> { background: lime; }
#\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\[\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\>\\\\+\\\\<\\\\<\\\\<\\\\<\\\\-\\\\]\\\\>\\\\+\\\\+\\\\.\\\\>\\\\+\\\\.\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\.\\\\+\\\\+\\\\+\\\\.\\\\>\\\\+\\\\+\\\\.\\\\<\\\\<\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\>\\\\.\\\\+\\\\+\\\\+\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\>\\\\+\\\\.\\\\>\\\\. { background: lime; }
#\\\\# { background: lime; }
#\\\\#\\\\# { background: lime; }
#\\\\#\\\\.\\\\#\\\\.\\\\# { background: lime; }
#\\\\_ { background: lime; }
#\\\\{\\\\} { background: lime; }
#\\\\.fake\\\\-class { background: lime; }
#foo\\\\.bar { background: lime; }
#\\\\3A hover { background: lime; }
#\\\\3A hover\\\\3A focus\\\\3A active { background: lime; }
#\\\\[attr\\\\=value\\\\] { background: lime; }
#f\\\\/o\\\\/o { background: lime; }
#f\\\\\\\\o\\\\\\\\o { background: lime; }
#f\\\\*o\\\\*o { background: lime; }
#f\\\\!o\\\\!o { background: lime; }
#f\\\\'o\\\\'o { background: lime; }
#f\\\\~o\\\\~o { background: lime; }
#f\\\\+o\\\\+o { background: lime; }

.♥ { background: lime; }
.© { background: lime; }
.“‘’” { background: lime; }
.☺☃ { background: lime; }
.⌘⌥ { background: lime; }
.𝄞♪♩♫♬ { background: lime; }
.💩 { background: lime; }
.\\\\? { background: lime; }
.\\\\@ { background: lime; }
.\\\\. { background: lime; }
.\\\\3A \\\\) { background: lime; }
.\\\\3A \\\\\`\\\\( { background: lime; }
.\\\\31 23 { background: lime; }
.\\\\31 a2b3c { background: lime; }
.\\\\<p\\\\> { background: lime; }
.\\\\<\\\\>\\\\<\\\\<\\\\<\\\\>\\\\>\\\\<\\\\> { background: lime; }
.\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\[\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\>\\\\+\\\\+\\\\+\\\\>\\\\+\\\\<\\\\<\\\\<\\\\<\\\\-\\\\]\\\\>\\\\+\\\\+\\\\.\\\\>\\\\+\\\\.\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\.\\\\+\\\\+\\\\+\\\\.\\\\>\\\\+\\\\+\\\\.\\\\<\\\\<\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\+\\\\.\\\\>\\\\.\\\\+\\\\+\\\\+\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\-\\\\.\\\\>\\\\+\\\\.\\\\>\\\\. { background: lime; }
.\\\\# { background: lime; }
.\\\\#\\\\# { background: lime; }
.\\\\#\\\\.\\\\#\\\\.\\\\# { background: lime; }
.\\\\_ { background: lime; }
.\\\\{\\\\} { background: lime; }
.\\\\#fake\\\\-id { background: lime; }
.foo\\\\.bar { background: lime; }
.\\\\3A hover { background: lime; }
.\\\\3A hover\\\\3A focus\\\\3A active { background: lime; }
.\\\\[attr\\\\=value\\\\] { background: lime; }
.f\\\\/o\\\\/o { background: lime; }
.f\\\\\\\\o\\\\\\\\o { background: lime; }
.f\\\\*o\\\\*o { background: lime; }
.f\\\\!o\\\\!o { background: lime; }
.f\\\\'o\\\\'o { background: lime; }
.f\\\\~o\\\\~o { background: lime; }
.f\\\\+o\\\\+o { background: lime; }
"
`;
